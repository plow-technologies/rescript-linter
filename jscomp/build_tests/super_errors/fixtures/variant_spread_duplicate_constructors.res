type a = One | Two
type b = Two | Three
type c = | ...a | ...b | Four
