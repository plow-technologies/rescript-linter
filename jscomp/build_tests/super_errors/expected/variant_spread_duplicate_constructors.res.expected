
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/variant_spread_duplicate_constructors.res[0m:[2m3:22[0m

  1 [2mâ”‚[0m type a = One | Two
  2 [2mâ”‚[0m type b = Two | Three
  [1;31m3[0m [2mâ”‚[0m type c = | ...a | ...[1;31mb[0m | Four
  4 [2mâ”‚[0m 

  Variant b has a constructor named Two, but a constructor named Two already exists in the variant it's spread into.
  You cannot spread variants with overlapping constructors.